tgui/packages/tgui/interfaces/ExoscannerConsole.tsx

```diff
@@ -0,0 +1,241 @@
+                <BlockQuote>Deep scan performs full scan of the site, revealing all details.</BlockQuote>
+                <Box><Button disabled={scan_power <= 0} onClick={() => act("start_deep_scan")}>Scan</Button> <Box inline pl={3}>Estimated Time: {deep_cost}.</Box></Box>
+              </Section>
+            )}
+          </Section>
+        </Stack.Item>
+      )}
+    </Stack>);
+};
+
+type ScanInProgressData = {
+  scan_time: number,
+  scan_power: number,
+  scan_description: string,
+}
+
+const ScanInProgressModal = (props, context) => {
+  const { act, data } = useBackend<ScanInProgressData>(context);
+  const {
+    scan_time,
+    scan_power,
+    scan_description,
+  } = data;
+
+  return (
+    <Modal ml={1}>
+      <NoticeBox>Scan in Progress!</NoticeBox>
+      <Box color="danger" />
+      <LabeledList>
+        <LabeledList.Item
+          label="Scan summary">
+          {scan_description}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Time left">
+          {formatTime(scan_time)}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Scanning array power">
+          {scan_power}
+        </LabeledList.Item>
+        <LabeledList.Item
+          label="Emergency Stop">
+          <Button.Confirm
+            content="STOP SCAN"
+            color="red"
+            icon="times"
+            onClick={() => act("stop_scan")} />
+        </LabeledList.Item>
+      </LabeledList>
+    </Modal>);
+};
+
+
+type ExoscannerConsoleData = {
+  scan_in_progress: boolean,
+  scan_power: number,
+  possible_sites: Array<SiteData>,
+  wide_scan_eta: number,
+  selected_site: string,
+  failed: boolean,
+  scan_conditions: Array<string>,
+}
+
+export const ExoscannerConsole = (props, context) => {
+  const { act, data } = useBackend<ExoscannerConsoleData>(context);
+  const {
+    scan_in_progress,
+    scan_power,
+    possible_sites = [],
+    wide_scan_eta,
+    selected_site,
+    failed,
+    scan_conditions = [],
+  } = data;
+
+  const can_start_wide_scan = scan_power > 0;
+
+  return (
+    <Window>
+      {!!scan_in_progress && (<ScanInProgressModal />)}
+      {!!failed && (<ScanFailedModal />)}
+      <Window.Content>
+        <Stack vertical fill>
+          <Stack.Item>
+            <Section fill title="Available array power">
+              <Stack>
+                <Stack.Item grow>
+                  {scan_power > 0 && (
+                    <>
+                      <Box pr={1} inline fontSize={2}>{scan_power}</Box>
+                      <Icon
+                        name="satellite-dish"
+                        size={3} />
+                    </>) || ("No properly configured scanner arrays detected.")}
+                </Stack.Item>
+              </Stack>
+              <Section title="Special Scan Condtions">
+                {scan_conditions && scan_conditions.map(condition => <NoticeBox key={condition} warning>{condition}</NoticeBox>)}
```

Comment:
tab out this map, look at how i showed how to tab out the other maps in this review